src/app/folder/folderlist.component.ts

```diff
@@ -393,7 +396,16 @@ export class FolderListComponent {
             sourceIndex++;
         }
 
-        // console.log('after rearrange', folders.map(f => `${f.folderId} ${f.folderPath}`));
+        // Adjust the priorities, client-side
+        // Eventually the backend will do it by itself, and we'll se the results on the next reload.
+        // We'd rather see an instant change in the UI though -- so let's calculate temporary priorities
+        // so that everythings looks alright.
+        let priority = 1;
+        for (const f of folders) {
```

Comment:
is this needed? when sent to the server below, the array of folder ids are sent in the order they appear. (priority field is only used for the client side sorting)

Reply:
It's not needed for the server: it's needed for the folderlist component. Few lines down we do [`this.messagelistservice.folderCountSubject.next(folders);`](https://github.com/runbox/runbox7/blob/04beb7e36a41cddc0495fec52fe5eda366a4c2c7/src/app/folder/folderlist.component.ts#L410), which goes straight to the UI [here](https://github.com/runbox/runbox7/blob/04beb7e36a41cddc0495fec52fe5eda366a4c2c7/src/app/folder/folderlist.component.ts#L94) before it goes to the API. The UI then re-renders them, and if this line here is missing, it takes old priorities into account more seriously than the order of entries, so after the drag and drop the UI on the screen remains the same as before: even though the API gets updated, and the next page load will show the folders after the change.

I hoped that the comment explains this, but perhaps it needs to be more detailed :)

Reply:
The reason it wasn't needed before is because the sorting happened in the API service rather than the UI, so when we were feeding the UI with new, after-drag-and-drop data, the re-sorting would not happen like it does now. It's a bit of a hack, but the alternative would be to teach the API service about the tree structure that we need, which feels a bit awkward when it's already doing the flattening of the tree it gets in the first place :­P

Reply:
OK - sounds good :-) Good that you're fixing this !

Reply:
Sweet :) Wanna give me your Seal of Approval? ;)